### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by TheKodeToad**
Refactor.
I decided it was best to stop now! xD

### CODE REVIEW

File: src/main/java/xyz/spruceloader/trunk/Trunk.java [hunk 1]
```diff
@@ -1,54 +1,63 @@
...
-    private static Launch INSTANCE;
+public class Trunk {
+    public static final boolean DEVELOPMENT = Boolean.getBoolean("trunk.development");
+    private static final Logger LOGGER = LoggerFactory.getLogger("Trunk");
 
-    private Map<String, Object> globalProperties = new HashMap<>();
-    private LaunchClassLoader classLoader;
+    private final TrunkClassLoader classLoader;
+    private final TransformerManager transformers;
+    private final Map<String, Object> globalProperties = new HashMap<>();
```
Comment (Deftu) [inline src/main/java/xyz/spruceloader/trunk/Trunk.java]: If there isn't going to be a singleton instance, it's probably best to make the global properties static.

File: src/main/java/xyz/spruceloader/trunk/Trunk.java [hunk 2]
```diff
@@ -1,54 +1,63 @@
...
+        }).filter(Objects::nonNull).collect(Collectors.toList()).toArray(URL[]::new), getClass().getClassLoader());
+        transformers = new TransformerManager();
         Thread.currentThread().setContextClassLoader(classLoader);
-        globalProperties.put("launch.development", DEVELOPMENT);
+        globalProperties.put("trunk.development", DEVELOPMENT);
     }
 
     public void initialize(ArgumentMap argMap, EnvSide env) {
-        LOGGER.info("Launching Minecraft with Spruce Launchwrapper");
+        LOGGER.info("Launching Minecraft with Trunk");
```
Comment (Deftu) [inline src/main/java/xyz/spruceloader/trunk/Trunk.java]: This should probably be changed to say "Spruce Trunk" to make it more transparent what Trunk actually is.

File: src/main/java/xyz/spruceloader/trunk/Trunk.java [hunk 3]
```diff
@@ -1,54 +1,63 @@
...
-    private static final Logger LOGGER = LoggerFactory.getLogger("Launchwrapper");
-    private static Launch INSTANCE;
+public class Trunk {
+    public static final boolean DEVELOPMENT = Boolean.getBoolean("trunk.development");
+    private static final Logger LOGGER = LoggerFactory.getLogger("Trunk");
 
-    private Map<String, Object> globalProperties = new HashMap<>();
-    private LaunchClassLoader classLoader;
+    private final TrunkClassLoader classLoader;
+    private final TransformerManager transformers;
```
Comment (Deftu) [inline src/main/java/xyz/spruceloader/trunk/Trunk.java]: If there's going to be a manager for transformers, it's probably best to name this field and it's getter according to the class name, as the class itself already has a `getTransformers()` method.

**Review by Deftu — CHANGES_REQUESTED** [review]
This is good to merge as soon as the problems above have been addressed

File: src/main/java/xyz/spruceloader/trunk/Trunk.java (hunk 1 above)
Comment (TheKodeToad) [reply src/main/java/xyz/spruceloader/trunk/Trunk.java]: Well, that would have issues if you had multiple instances.

Comment (Deftu) [reply src/main/java/xyz/spruceloader/trunk/Trunk.java]: Well there's no reason to have "global" properties if they can only be accessed within transformers - which is the only way to retrieve an instance atcm.

**Review by Deftu — APPROVED** [review]

**Pull request merged**